telephony/android_modem.c

```diff
@@ -3351,12 +3364,25 @@ amodem_addTimeUpdate( AModem  modem )
     * and deal with this case (since it normally relied on the operator's country code
     * which is hard to simulate on a general-purpose computer
     */
-    amodem_add_line( modem, "%%CTZV: %02d/%02d/%02d:%02d:%02d:%02d%c%d:%d:%s\r\n",
-             (utc.tm_year + 1900) % 100, utc.tm_mon + 1, utc.tm_mday,
-             utc.tm_hour, utc.tm_min, utc.tm_sec,
-             (tzdiff >= 0) ? '+' : '-', (tzdiff >= 0 ? tzdiff : -tzdiff) / 15,
-             (local.tm_isdst > 0),
-             tzname );
+    amodem_add_cvtz_line(modem, &utc, &local, tzdiff, tzname);
+}
+
+/**
+ * Update the NITZ timezone to tzdiff from UTC in number of quarter-hours.
+ */
+void
+amodem_set_timezone( AModem modem, int tzdiff )
+{
+    time_t       utc_now = time(NULL);
+    time_t       local_now = utc_now + tzdiff * 15 * 60;
+    struct tm    utc, local;
+
+    utc   = *gmtime( &utc_now );
+    local = *gmtime( &local_now );
+
+    amodem_begin_line( modem );
+    amodem_add_cvtz_line(modem, &utc, &local, tzdiff, "Unknown/Unknown");
+    amodem_end_line_reply( modem );
```

Comment:
Since this is an unsolicited response, use `amodem_end_line_unsol()` which won't append an extra "OK" string.